templates/items/details/details-spell.hbs

```diff
@@ -0,0 +1,232 @@
+            {{ formField fields.materials.fields.supply value=source.materials.supply label="DND5E.Supply" localize=true
+                         classes="label-top" placeholder="0" }}
+
+            {{!-- Material Cost --}}
+            <div class="form-group label-top">
+                <label class="label-icon currency gp" aria-label="{{ localize "DND5E.CostGP" }}">
+                    {{ localize "DND5E.Cost" }}
+                </label>
+                <div class="form-fields">
+                    {{ formInput fields.materials.fields.cost value=source.materials.cost placeholder="—" }}
+                </div>
+            </div>
+
+            {{!-- Material Consumption --}}
+            {{ formField fields.materials.fields.consumed value=source.materials.consumed label="DND5E.Consumed"
+                         localize=true input=inputs.createCheckboxInput classes="checkbox" }}
+        </div>
+
+        {{!-- Materials Description --}}
+        {{ formInput fields.materials.fields.value value=source.materials.value input=inputs.createTextInput
+                     classes="full-width" }}
+    </div>
+    {{/if}}
+
+    {{!-- Preparation Mode --}}
+    <div class="form-group">
+        <label>{{ localize "DND5E.SpellPreparation" }}</label>
+        <div class="form-fields">
+
+            {{!-- Prepared --}}
+            {{#if (eq source.preparation.mode "prepared")}}
+            {{ formInput fields.preparation.fields.prepared value=source.preparation.prepared
+                         dataset=(dnd5e-object tooltip="DND5E.Prepared") input=inputs.createCheckboxInput
+                         ariaLabel=(localize "DND5E.Prepared") }}
+            {{/if}}
+
+            {{!-- Mode --}}
+            {{ formInput fields.preparation.fields.mode value=source.preparation.mode
+                         choices=config.spellPreparationModes labelAttr="label" }}
+        </div>
+    </div>
+
+    {{!-- Source Class --}}
+    {{#if isEmbedded}}
+    {{ formField fields.sourceClass value=source.sourceClass localize=true choices=item.parent.spellcastingClasses
+                 labelAttr="name" blank="" }}
+    {{/if}}
+</fieldset>
+
+<fieldset>
+    <legend>{{ localize "DND5E.Casting" }}</legend>
+
+    {{!-- Activation --}}
+    <div class="form-group split-group">
+        <label>{{ localize "DND5E.SpellCastTime" }}</label>
+        <div class="form-fields">
+
+            {{!-- Amount --}}
+            {{#if (and source.activation.type (not isCostlessAction))}}
+                {{ formField fields.activation.fields.value value=source.activation.value label="DND5E.Amount" placeholder="—"
+                             localize=true hint=false classes="label-top" }}
+            {{/if}}
+
+            {{!-- Type --}}
+            {{ formField fields.activation.fields.type value=source.activation.type label="DND5E.Cost" localize=true
+                         classes="label-top" hint=false choices=config.abilityActivationTypes blank="" }}
```

Comment:
We'll want to switch this over to using the new `CONFIG.DND5E.activityActivationTypes` with grouping.